### ISSUE TITLE
fix(training): test_combined_audio_objective_overfits_one_fixed_online_example red on main

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#3138]
`tests/models/test_vst_flow_audio_feedback.py::test_combined_audio_objective_overfits_one_fixed_online_example` fails on current `main` (d69c855ea2):

```
assert 0.23520737886428833 < 0.2
```

- Deterministic: the same loss value reproduces bitwise across runs, solo or in-suite, on a clean checkout of `origin/main` (RTX 5060 Ti host).
- Passes at c51ff04d1d, fails at d69c855ea2. The only `src/synth_setter/models` change in that window is #3123 (`internal-feat(training): support temporal reverb sketch tokenization`, merged as 2ce77f639f), so the drift likely entered there — possibly an RNG-consumption-order shift in this seeded overfit test rather than a real regression, but that needs confirming.

Repro: `uv run pytest tests/models/test_vst_flow_audio_feedback.py::test_combined_audio_objective_overfits_one_fixed_online_example -q`

Expected: loss overfits below the 0.2 threshold as before.

**Comment by ktinubu** [comment]
Nightly data point: the same test fails on every Nightly Tests run since 2026-09-05 at head a0bbc99 with an identical value, `assert 0.48064136505126953 < (2.73732852935791 * 0.1)`, so in the full nightly session the ratio check fails too, not just the absolute 0.2 floor. On the same commit the CPU Slow Tests lane (slow-marker subset, separate session) passes it, and a solo local run lands at 0.235. The final loss therefore depends on session-level state or platform numerics, which points at test-order or thread-count sensitivity rather than a single regression. Left out of the CI-green PR bundle (#3156, #3157, #3158) because tuning the threshold blind would hide that.

**Comment by ktinubu** [comment]
Counter data point: a `workflow_dispatch` of Nightly Tests on the #3157 branch (same `main` head a0bbc99 plus only an rclone install step, run 34224499144) passed this test; the run's sole failure was the mutmut shard test. So the value flips between 0.48 (four consecutive scheduled nightlies) and a pass on the same code and workflow. Runner hardware or scheduling differences between runs change the optimizer trajectory, so the fix should target determinism or the threshold margin, not a specific regression commit.

**Comment by ktinubu** [comment]
Local sweep on the test's own helpers (seed 0, lr 3e-4, one audible example), varying only `torch.set_num_threads`:

| threads | steps | final | mean of last 50 | max of last 50 | min over run |
| --- | --- | --- | --- | --- | --- |
| 1 | 300 | 0.054 | 0.115 | 0.188 | 0.052 |
| 4 | 300 | 0.336 | 0.326 | 0.561 | 0.034 |
| 1 | 600 | 0.128 | | | |
| 4 | 600 | 0.025 | | | |
| 1 | 1000 | 0.094 | | | |
| 4 | 1000 | 0.096 | | | |

Initial total is 2.737 in every run. The trajectory reaches well under 0.1 in all of them but then oscillates; where the last step lands depends on the BLAS reduction order (thread count) and is not monotonic in step count. The CI values seen so far (0.105, 0.24, 0.48, and passes) are samples of that oscillation. A final-step threshold cannot be made stable by moving it. Options: assert on the best loss reached (weaker "reachable" claim, stable at 0.03-0.05 here), or damp the trajectory (lower lr with a short cosine decay, or Adam with a smaller final lr) so the last step is representative. Either is a training-semantics decision, so not folded into the CI-green bundle. This test now also fails the PR slow lane on #3156, so it blocks readiness for any PR that runs `run_slow_tests`.
